Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py

```diff
@@ -0,0 +1,248 @@
+match = re.match(pattern, str(logAnalyticsUri))
+if (not match):
+    raise Exception("Lookout: Invalid Log Analytics Uri.")
+
+##############################
+######State Manager######  
+##############################
+class StateManager:
+    def __init__(self, connection_string, share_name='funcstatemarkershare', file_path='Anomalyfuncmarkerfile'):
+        self.share_cli = ShareClient.from_connection_string(conn_str=connection_string, share_name=share_name)
+        self.file_cli = ShareFileClient.from_connection_string(conn_str=connection_string, share_name=share_name, file_path=file_path)
+
+    def post(self, marker_text: str):
+        try:
+            self.file_cli.upload_file(marker_text)
+        except ResourceNotFoundError:
+            self.share_cli.create_share()
+            self.file_cli.upload_file(marker_text)
+
+    def get(self):
+        try:
+            return self.file_cli.download_file().readall().decode()
+        except ResourceNotFoundError:
+            return None
+##############################
+######lookout Connector######  
+##############################
+
+class LookOut:
+
+    def __init__(self):
+        self.api_key = jwt_api_key
+        self.api_secret = jwt_api_secret
+        self.base_url = baseurl
+        self.jwt_token_exp_hours = 1
+        self.jwt_token = self.get_new_token()          
+
+    def get_new_token(self):
+        url = Authurl
+        payload = json.dumps({
+                "clientId": self.api_key,
+                "clientSecret": self.api_secret,
+                "grant_type": "refresh_token"
+                })
+        headers = {
+                'Content-Type': 'application/json'
+                }
+        response = requests.request("POST", url, headers=headers, data=payload)
+        tokens = json.loads(response.text) 
+        return tokens['id_token']        
+	    
+    def generate_date(self):
+        current_time_day = datetime.datetime.utcnow().replace(second=0, microsecond=0) 
+        logging.info("Present time {}".format(current_time_day))
+        current_time_day = (current_time_day - datetime.timedelta(minutes=int(fetchDelay))).strftime("%Y-%m-%dT%H:%M:%S")+".000-00:00"       
+        logging.info("the fetch delay taken as {} minutes".format(fetchDelay))
+        logging.info("After fetch time {}".format(current_time_day))
+        state = StateManager(connection_string)
+        past_time = state.get()
+        if past_time is not None:
+            logging.info("The last time run happened at: {}".format(past_time))
+        else:
+            logging.info("There is no last run timestamp, trying to get Anamolies for last week.")
+            logging.info("The past days were taken as {} days".format(pastDays))
+            past_time = (datetime.datetime.utcnow().replace(second=0, microsecond=0) - datetime.timedelta(days=int(pastDays))).strftime("%Y-%m-%dT%H:%M:%S")+".000-00:00"
+        state.post(current_time_day)
```

Comment:
1. I think we need to remove this statement. We should post only if we don't see any exceptions. In case of exceptions, time should not get incremented.

Reply:
done.